packages/notifier/src/notifier.js

```diff
@@ -1,42 +1,78 @@
-/* global harden */
+// @ts-check
+// eslint-disable-next-line spaced-comment
+/// <reference types="ses"/>
 
 import { producePromise } from '@agoric/produce-promise';
 
+/**
+ * @typedef {Object} UpdateHandle a value used to mark the position in the update stream
+ */
+
+/**
+ * @template T the type of the state value
+ * @typedef {Object} UpdateRecord<T>
+ * @property {T} value is whatever state the service wants to publish
+ * @property {UpdateHandle} updateHandle is a value that identifies the update
+ * @property {boolean} done false until the updater publishes a final state
+ */
+
+/**
+ * @template T the type of the notifier state
+ * @callback GetUpdateSince<T> Can be called repeatedly to get a sequence of update records
+ * @param {UpdateHandle} [updateHandle] return update record as of a handle
```

Comment:
Shouldn't this have a question mark or other marker indicating that it's optional?

Reply:
The `[updateHandle]` indicates that it's optional.